cmd/argo/commands/client/conn.go

```diff
@@ -54,6 +54,11 @@ func AddAPIClientFlagsToCmd(cmd *cobra.Command) {
 	cmd.PersistentFlags().BoolVarP(&ArgoServerOpts.InsecureSkipVerify, "insecure-skip-verify", "k", os.Getenv("ARGO_INSECURE_SKIP_VERIFY") == "true", "If true, the Argo Server's certificate will not be checked for validity. This will make your HTTPS connections insecure. Defaults to the ARGO_INSECURE_SKIP_VERIFY environment variable.")
 }
 
+func AddKubectlFlagsToArgoServerOpts() {
+	ArgoServerOpts.ClientCert = overrides.AuthInfo.ClientCertificate
+	ArgoServerOpts.ClientKey = overrides.AuthInfo.ClientKey
+}
```

Comment:
I think this would make more sense to put right below `AddKubectlFlagsToCmd` to keep the `kubectl` inheritance together

we probably also want to add a comment here, something like `// some kubectl flags apply to server mode as well`

Reply:
Indeed, that is much better I was also unsure if it is a good idea to create a new `func`. I will adjust it

Reply:
The separate function is fine, especially as it is a side-effect, but putting the two functions together in the same file helps keep the functionality together

Reply:
So for the above, I noticed the following, I had to use pointers to string cause we call this during the `root` `cmd` the `overrides.AuthInfo` is still empty. If we do not want to use pointers here I can move the `AddKubectlFlagsToArgoServerOpts()` call inside the ``NewApiClient` func.

Reply:
did you retest to make sure this works? my only concern is the ordering such that `overrides` is no longer empty by the time it's called

Reply:
Yeah I did a couple of retests and I could see the certificates, also I could see them on the debug logs for argoserver opts, when before it wasn't there. Could also see the handshake at the nginx logs.

Reply:
Great! Just wanted to make sure since the ordering in the CLI can be a bit hard to trace